src/views/Editor.vue

```diff
@@ -30,36 +31,30 @@ const proposalData = computed(() => {
 });
 
 if (!proposals[proposalKey]) {
-  proposals[proposalKey] = {
-    title: '',
-    body: '',
-    discussion: '',
-    execution: [],
-    updatedAt: Date.now()
-  };
+  createDraft(id, { id: key });
 }
 
-if (!proposals[proposalKey].execution) proposals[proposalKey].execution = [];
-
 onMounted(() => {
   spacesStore.fetchSpace(spaceId, networkId as NetworkID);
 
   if (!key && route.name) {
     globalModalOpen.value = false;
 
-    const str = (Math.random() + 1).toString(36).substring(7);
+    const draftId = createDraft(id);
+
     router.replace({
       name: route.name,
-      params: { id, key: str }
+      params: { id, key: draftId }
     });
   }
 });
 
-watch(proposals, () => {
-  if (!proposals[proposalKey]) {
-    router.replace({ name: 'editor' });
-  }
-});
+// TODO we dont need it cos we handle !key in onMounted and router.replace() there
+// watch(proposals, () => {
+//   if (!proposals[proposalKey]) {
+//     router.replace({ name: 'editor' });
+//   }
+// });
```

Comment:
This handles case when draft gets deleted (if draft gets deleted, navigate away from it):

https://user-images.githubusercontent.com/1968722/217051700-ba0a17cb-0b54-4e54-b8b0-3aa34aa723c9.mp4

Reply:
So maybe we watch for specific case in watcher?
```
watch(
  () => proposals[proposalKey],
  (to, from) => {
    if (from && !to) {
      router.replace({ name: 'editor' });
    }
  }
);
```

Reply:
If that works then it sounds good to me.